src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -117,6 +117,16 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// </summary>
         bool IsReadOnly { get; }
 
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+
+        /// <summary>
+        /// Converts an <c>ITypeSymbol</c> and a nullable annotation to a string representation.
+        /// </summary>
+        /// <param name="topLevelNullability">The top-level nullability to use for formatting.</param>
+        /// <param name="format">Format or null for the default.</param>
+        /// <returns>A formatted string representation of the symbol with the given nullability.</returns>
+        string ToDisplayString(NullableAnnotation topLevelNullability, SymbolDisplayFormat format = null);
```

Comment:
If you need to pass the symbol and these enum values around, can they just be embedded with the symbol.  that way you get the right behavior by default without needing to do all of this. 

I believe the compiler has that concept internally (TypeWithAnnotations or something like that).

Reply:
It does. @333fred where are the design meeting notes when we hashed this out a bunch of times?

Reply:
Ah, right the general notes are [here](https://github.com/dotnet/roslyn/blob/master/docs/compilers/CSharp/Nullability%20Public%20API%20Design%20Notes.md) but I'm not sure if we discussed the struct specifically.

Reply:
Ick, based on the last time i talked to fred, i thought this would *not* be happening.  There was ample precedent and evidence this would be highly unpleasant.  First, it makes whole swaths of our APIs incorrect (like so much of the code in SyntaxGenerator). Second, it means having to manually investigate and update effectively all features.  Finally, the reasoning for this was specious (i.e. memory savings) as we have ample evidence from projects like TypeScript that memory usage is fine here**

TypeScript has to go even further than we do.  Not only do they have core types like `string`, but they'll also have `string | undefined`, `string | null`, `string | undefined | null`?  Despite the need for all tehse variations of types (and many more), memory usage is fine.

Reply:
To close this conversation: this PR has spawned (quite a bit) of further discussion. I'm going to go with this PR as written only because the new APIs here will let us continue to push on other stuff and understand how this model either helps or hurts us for other areas.